### ISSUE TITLE
need an easy way to tool the choice of @Service ID when configuring components/endpoints

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8#3542]
so that if a user is running on kubernetes/jube, by default they choose or type the service ID to use - and thats use to generate the setters on the component/endpoint in the generated code.

See #3437 for the background.

e.g. we need to know which properties on a component/endpoint map to a service host / port so that we can present a service ID property as the first, mandatory item in the form (and hide the host/port fields) when in kubernetes/jube mode.

Then we'd generate code something like...
```
public class MyCamelProduces {
  // Note this was generated by Forge
  @Produces @Named("foo") @Singleton
  JmsComponent createFooComponent(@Service("MY_JMS_SERVICE") URL serviceURL) {
    ActiveMQComponent answer = new ActiveMQComponent();
    answer.setSomethingHost(serviceURL.getHost());
    answer.setSomethingPortt(serviceURL.getPort());

    // other customisations go here...
    return answer;
  }
}
```

**Comment by davsclaus** [comment]
Yeah I think we need some kind of kubernetes service tooling in forge / fabric8. 
Also to adjust the pom.xml etc.

**Comment by davsclaus** [comment]
So if I develop a Camel app and then i add a jetty endpoint because I want this app to expose a service others can use.

I would then need to define that service in the pom.xml so the kube json has it included. And maybe I need that jetty endpoint service defined in the cdi component of jetty / or from <bean> in a xml file or from spring java config or something.

**Comment by davsclaus** [comment]
So down the road should we introduce some service api in Camel that components can optional implement so we can inject the service url, and the component knows how to configure from it, eg set host / port.

eg the thingy you have mentioned in the past.
